bionlp.py

```diff
@@ -0,0 +1,113 @@
+from transformers import pipeline
+import pandas as pd
+import requests
+from tqdm import tqdm
+
+def process_text(text):    
+    tdf1 = __tag_genes(text)
+    tdf2 = __tag_chemicals(text)
+    tdf3 = __tag_diseases(text)
+    df = pd.concat([tdf1,tdf2,tdf3])
+    df = __drop_unknowns(df)
+    df = normalize(df)
+    return df
+
+def batch(corpus):
+    df = pd.DataFrame()
+    for entry in corpus:
+        tdf1 = __tag_genes(entry)
+        tdf2 = __tag_chemicals(entry)
+        tdf3 = __tag_diseases(entry)
+        df = pd.concat([df,tdf1,tdf2,tdf3])
+    df = __drop_unknowns(df)
+    df['start'] = df['start'].astype(int)
+    df['end'] = df['end'].astype(int)
+    df = normalize(df)
+    return df
+
+def normalize(result):
+    result['concept_match_type'] = None
+    result['concept_id'] = None
+    result['concept_label'] = None
+    for index, row in tqdm(result.iterrows()):
+        if row['entity_group'] == 'GENETIC':
+            norm_result = __normalize_gene(row['word'])
+        if row['entity_group'] == 'CHEMICAL':
+            norm_result = __normalize_therapy(row['word'])
+        if row['entity_group'] == 'DISEASE':
+            norm_result = __normalize_disease(row['word'])
+        result.at[index, 'concept_match_type'] = norm_result[0]
+        result.at[index, 'concept_id'] = norm_result[1]
+        result.at[index, 'concept_label'] = norm_result[2]
+    return result
+
+def __normalize_gene(word):
+    r = requests.get(f'https://normalize.cancervariants.org/gene/normalize?q={word}')
```

Comment:
fair warning, this API is gonna change soonish

Reply:
🫡